tools/crm_resource_runtime.c

```diff
@@ -662,6 +662,24 @@ cli_resource_delete_failures(crm_ipc_t *crmd_channel, const char *host_uname,
 {
     int rc = pcmk_ok;
 
+    if (rsc == NULL) {
+        return -ENXIO;
```

Comment:
It's OK to call --cleanup without a resource. That's why we only filter if the user specified rsc_id:

    } else if(rsc_id && safe_str_neq(rsc_id, resource_name)) {
        continue;

Earlier, in:

    if (find_flags && rsc_id) {

we already returned -ENXIO if the user specified rsc_id but that resource doesn't exist.

Reply:
Makes sense.